.goreleaser.yml

```diff
@@ -0,0 +1,132 @@
+    asmflags:
+      - all=-trimpath={{ dir .ArtifactPath | dir | dir | dir }}
+    gcflags:
+      - all=-trimpath={{ dir .ArtifactPath | dir | dir | dir }}
+    ldflags:
+      - -X {{ .Env.REPO }}/internal/version.Version={{ if not .IsSnapshot }}v{{ end }}{{ .Version }}
+      - -X {{ .Env.REPO }}/internal/version.GitVersion={{ .Tag }}
+      - -X {{ .Env.REPO }}/internal/version.GitCommit={{ .FullCommit }}
+      - -X {{ .Env.REPO }}/internal/version.KubernetesVersion={{ .Env.K8S_VERSION }}
+    goarch:
+      - amd64
+      - arm64
+      - ppc64le
+      - s390x
+    goos:
+      - linux
+      - darwin
+    ignore:
+      - goos: darwin
+        goarch: arm64
+      - goos: darwin
+        goarch: ppc64le
+      - goos: darwin
+        goarch: s390x
+
+  # ansible-operator build steps
+  - id: ansible-operator
+    main: ./cmd/ansible-operator
+    binary: ansible-operator
+    env:
+      - CGO_ENABLED=0
+      - GO111MODULE=on
+      - REPO=github.com/operator-framework/operator-sdk
+    mod_timestamp: "{{ .CommitTimestamp }}"
+    asmflags:
+      - all=-trimpath={{ dir .ArtifactPath | dir | dir | dir }}
+    gcflags:
+      - all=-trimpath={{ dir .ArtifactPath | dir | dir | dir }}
+    ldflags:
+      - -X {{ .Env.REPO }}/internal/version.Version={{ if not .IsSnapshot }}v{{ end }}{{ .Version }}
+      - -X {{ .Env.REPO }}/internal/version.GitVersion={{ .Tag }}
+      - -X {{ .Env.REPO }}/internal/version.GitCommit={{ .FullCommit }}
+      - -X {{ .Env.REPO }}/internal/version.KubernetesVersion={{ .Env.K8S_VERSION }}
+    goarch:
+      - amd64
+      - arm64
+      - ppc64le
+      - s390x
+    goos:
+      - linux
+      - darwin
+    ignore:
+      - goos: darwin
+        goarch: arm64
+      - goos: darwin
+        goarch: ppc64le
+      - goos: darwin
+        goarch: s390x
+
+  # helm-operator build steps
+  - id: helm-operator
+    main: ./cmd/helm-operator
+    binary: helm-operator
+    env:
+      - CGO_ENABLED=0
+      - GO111MODULE=on
+      - REPO=github.com/operator-framework/operator-sdk
+    mod_timestamp: "{{ .CommitTimestamp }}"
+    asmflags:
+      - all=-trimpath={{ dir .ArtifactPath | dir | dir | dir }}
+    gcflags:
+      - all=-trimpath={{ dir .ArtifactPath | dir | dir | dir }}
+    ldflags:
+      - -X {{ .Env.REPO }}/internal/version.Version={{ if not .IsSnapshot }}v{{ end }}{{ .Version }}
+      - -X {{ .Env.REPO }}/internal/version.GitVersion={{ .Tag }}
+      - -X {{ .Env.REPO }}/internal/version.GitCommit={{ .FullCommit }}
+      - -X {{ .Env.REPO }}/internal/version.KubernetesVersion={{ .Env.K8S_VERSION }}
+    goarch:
+      - amd64
+      - arm64
+      - ppc64le
+      - s390x
+    goos:
+      - linux
+      - darwin
+    ignore:
+      - goos: darwin
+        goarch: arm64
+      - goos: darwin
+        goarch: ppc64le
+      - goos: darwin
+        goarch: s390x
+
+# Use most recent tag and short commit for snapshot version.
+snapshot:
+  name_template: "{{ .Tag }}"
```

Comment:
In my experimenting, I ended up using `.Env.GIT_VERSION` for both `snapshot` and `release` name_template. Go releaser seems to do some really odd `v` stripping for snapshots but not releases.

I was trying to make snapshot and non-snapshot runs look as close to the same thing as possible from a config standpoint.